### DISCUSSION TITLE
Reader fails in Dynamic Route inside app router root directory on production

### DISCUSSION THREAD

**Discussion by bastienrobert** [Thinkmill/keystatic#1387] [Q&A]
When trying to use the Keystatic reader to fetch content within a dynamic route under the app/ directory, the reader does not function as expected. 
This issue occurs specifically when the route is dynamic ([slug], [id], etc.), and the Keystatic reader is called to fetch content dynamically based on the route parameters.

**Steps to Reproduce:**
1. Set up a Next.js project using the app/ directory structure.
2. Install and configure Keystatic to manage static content (e.g., Markdown files).
3. Create a dynamic route, such as app/[slug]/page.tsx, and call the Keystatic reader inside it to fetch content based on the slug.
4. Try accessing the route with different dynamic values (/post-1, /post-2, etc.).

- https://nextjs-keystatic-test-params-repro.vercel.app/ (WORKING EXEMPLE)
- https://nextjs-keystatic-test-params-repro.vercel.app/fr (SAME CODE BUT IN app/[locale])

**REPO**
https://github.com/bastienrobert/nextjs-keystatic-test-params-repro

**Workaround:**
Dynamic import using the standard `import()` method, put every mdx and markdoc inline and format files to JSON.

**Comment by brooksbecton** [comment]
I'm not so sure if it is a bug with keystatic or nextjs. 
Maybe just innaccurate docs

If you follow the docs as is right now. The single collection example will be a dynamic route in NextJS.
To fix this issue, you'll need to statically render that single collection with the `generateStaticParams`

You can do that in Next 14 with 
```js
export async function generateStaticParams() {
  const posts = await reader.collections.posts.all();

  return posts.map((post) => ({ slug: post.slug }));
}
```

So maybe this part needs to be updated
https://keystatic.com/docs/installation-next-js#displaying-a-single-collection-entry
